.github/workflows/python-ci.yml

```diff
@@ -43,9 +43,14 @@ jobs:
 
     steps:
     - uses: actions/checkout@v3
+    - name: Install poetry
+      run: pip install poetry
```

Comment:
I'm not a fan of pinning development requirements unless you have information it will break anyway. This way the CI will learn us when Poetry isn't backwards compatible anymore. 

Remember to pin Poetry in the Makefile if decide to stick to a certain Poetry verison https://github.com/apache/iceberg/blob/abcc65b4f2e25dc9830cf0def907540fe15f5f76/python/Makefile#L19 also.